### PR TITLE
Have `clang-format` indent conditional CPP

### PR DESCRIPTION
**Pull request by Ericson2314**
# Motivation

This reflects the style I've been introducing in tandem with Windows support.

# Context

See https://clang.llvm.org/docs/ClangFormatStyleOptions.html#indentppdirectives for this configuration option.

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -28,3 +28,4 @@ EmptyLineBeforeAccessModifier: Leave
 #PackConstructorInitializers: BinPack
 BreakBeforeBinaryOperators: NonAssignment
 AlwaysBreakBeforeMultilineStrings: true
+IndentPPDirectives: AfterHash
```
Comment (jlesquembre) [inline .clang-format:31]: Maybe we should also consider setting `PPIndentWidth`: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#ppindentwidth

```suggestion
IndentPPDirectives: AfterHash
PPIndentWidth: 1
```

Comment (Ericson2314) [reply .clang-format:31]: I think two is better, I was doing 1 the first time to "get back on" even character offsets, but if it thinks it is better to do 1, 3, 5, etc. I am OK with that.

Comment (jlesquembre) [reply .clang-format:31]: In that case, let's go with 2. I don't have a preference; I only suggested 1 because I saw you went with one on #10486

Comment (Ericson2314) [reply .clang-format:31]:
```suggestion
IndentPPDirectives: AfterHash
PPIndentWidth: 2
```
OK sounds good

**Review by jlesquembre — APPROVED** [review]

**Pull request merged**